pkg/oc/cli/cmd/set/volume.go

```diff
@@ -699,41 +699,60 @@ func (v *VolumeOptions) getVolumeName(spec *kapi.PodSpec, singleResource bool) (
 				for _, m := range c.VolumeMounts {
 					if path.Clean(m.MountPath) == path.Clean(opts.MountPath) {
 						name = m.Name
-						matchCount += 1
+						matchCount++
 						break
 					}
 				}
 			}
 
 			switch matchCount {
 			case 0:
-				return "", fmt.Errorf("unable to find the volume for mount-path: %s", opts.MountPath)
+				return "", false, fmt.Errorf("unable to find the volume for mount-path: %s", opts.MountPath)
 			case 1:
-				return name, nil
+				return name, false, nil
 			default:
-				return "", fmt.Errorf("found multiple volumes with same mount-path: %s", opts.MountPath)
+				return "", false, fmt.Errorf("found multiple volumes with same mount-path: %s", opts.MountPath)
 			}
-		} else {
-			return "", fmt.Errorf("ambiguous --overwrite, specify --name or --mount-path")
 		}
-	} else { // Generate volume name
-		name := names.SimpleNameGenerator.GenerateName(volumePrefix)
-		if len(v.Output) == 0 {
-			fmt.Fprintf(v.Err, "info: Generated volume name: %s\n", name)
+		return "", false, fmt.Errorf("ambiguous --overwrite, specify --name or --mount-path")
+	}
+
+	oldName, claimFound := v.checkForExistingClaim(spec)
+
+	if claimFound {
+		return oldName, true, nil
+	}
+	// Generate volume name
+	name := names.SimpleNameGenerator.GenerateName(volumePrefix)
+	if len(v.Output) == 0 {
+		fmt.Fprintf(v.Err, "info: Generated volume name: %s\n", name)
+	}
+	return name, false, nil
+}
+
+func (v *VolumeOptions) checkForExistingClaim(spec *kapi.PodSpec) (string, bool) {
+	for _, vol := range spec.Volumes {
+		oldSource := vol.VolumeSource.PersistentVolumeClaim
+		if oldSource != nil && v.AddOpts.ClaimName == oldSource.ClaimName {
```

Comment:
no namespace?

Reply:
There is no need of checking for namespace here. Both PVC and Pod will belong to same namespace.